src/averageAges.js

```diff
@@ -55,7 +78,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let kids = people.filter(kid => kid.mother
+    && people.find(human => human.name === kid.mother));
+
+  kids = !onlyWithSon
+    ? kids
+    : kids.filter(kid => kid.sex === 'm');
```

Comment:
No need to use filter twice if `onlyWithSon` is passed, use it once and write needed condition inside